### ISSUE TITLE
Total refactoring

### ISSUE THREAD

**Issue by antonkomarev** [cybercog/laravel-ownership#21]
This package requires total refactoring before adding new Laravel versions support and new features.

**Comment by minitia** [comment]
hi @antonkomarev, what will it breaks if you remove the requirement of laravel 5? 
Basically they are traits attached to models (I'm not expert in Laravel)
Thanks in advance

**Comment by antonkomarev** [comment]
In theory L6 and L7 support could be added without removing L5 support.

**Comment by minitia** [comment]
this in you composer
"illuminate/database": "~5.2|~5.3|~5.4|~5.5|~5.6|~5.7",
"illuminate/support": "~5.2|~5.3|~5.4|~5.5|~5.6|~5.7"
doesn't block laravel to v5?

I'm trying to install your branch add-latest-laravel-versions-support

**Comment by antonkomarev** [comment]
Yes. I haven't added latest Laravel versions because this package was archived for a long time. You could send a PR for the new Laravel versions support.

**Comment by aaly00** [comment]
@antonkomarev does the laravel7 branch work with laravel 7, or does the package still need to be updated to laravel7

**Comment by antonkomarev** [comment]
This package is not working with latest laravel versions. I'm not using it on production anymore. It requires some refactoring.

**Comment by jny986** [comment]
Is this package still needed "Total Refactoring" to be an ongoing concern?
We are currently getting a "package is abandoned" message when installing the latest version.
I am happy to help put with the development that you are wanting to do if we can put together a plan of the development that is needed/wanted.
Thanks

**Comment by antonkomarev** [comment]
@jny986 I've removed abandoned message. Package was updated at the end of the 2020, but I forgot to remove this notice.
I don't have plans to continue actively maintaining it by myself, but if you have an ideas how it could be improved - it's always opened for PRs.

**Comment by aaly00** [comment]
I don't mind helping with the development as well.> Is this package still needed "Total Refactoring" to be an ongoing concern?
> 
> We are currently getting a "package is abandoned" message when installing the latest version.
> 
> I am happy to help put with the development that you are wanting to do if we can put together a plan of the development that is needed/wanted.
> 
> Thanks

**Comment by antonkomarev** [comment]
@aaly00 not really, if you are fine with current implementation - we could continue development without changing its architecture and create separate package with the new one requirements when needed. It will be awesome if you will write how are you using this package and what you want to improve.

**Comment by andrekutianski** [comment]
> This package requires total refactoring before adding new Laravel versions support and new features.

Supports Laravel 8/9?

**Comment by antonkomarev** [comment]
@andrekutianski Laravel 8 is supported. You may provide a PR with Laravel 9 support if you want.
